src/performanceMetric.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * This file is part of the Game Closure SDK.
+ *
+ * The Game Closure SDK is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public License v. 2.0 as published by Mozilla.
+
+ * The Game Closure SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public License v. 2.0 for more details.
+
+ * You should have received a copy of the Mozilla Public License v. 2.0
+ * along with the Game Closure SDK.  If not, see <http://mozilla.org/MPL/2.0/>.
+ */
+
+ /**
+ * @module performanceMetric
+ *
+ * Measures performance by keeping a window of delta times and providing a score using that window.
+ *
+**/
+
+var engine = null;
+var floor = Math.floor;
+
+var DEFAULT_RANK = 0;
+var DEFAULT_ALLOW_REDUCTION = true;
+
+	
+var PerformanceTester = Class(function () {
+	this.historySize = 200;
+	var _lastTick = Date.now();
+	var _history = [];
+	var historyIndex = 0;
+
+	this.init = function () {
+		this.LOWER_BOUND = 10;
```

Comment:
Let's change this API - these can be private variables with only a setter function. They are internal to this algorithm but we want to be able to set them. `var minFPS` and `var maxFPS` with a `this.setTargetFPSRange(min, max)` function